znvis/visualizer/headless_renderer.py

```diff
@@ -0,0 +1,284 @@
+"""
+ZnVis: A Zincwarecode package.
+License
+-------
+This program and the accompanying materials are made available under the terms
+of the Eclipse Public License v2.0 which accompanies this distribution, and is
+available at https://www.eclipse.org/legal/epl-v20.html
+SPDX-License-Identifier: EPL-2.0
+Copyright Contributors to the Zincwarecode Project.
+Contact Information
+-------------------
+email: zincwarecode@gmail.com
+github: https://github.com/zincware
+web: https://zincwarecode.com/
+Citation
+--------
+If you use this module please cite us with:
+
+Summary
+-------
+Main visualizer class.
+"""
+
+import os
+
+os.environ["OPENCV_IO_ENABLE_OPENEXR"] = "1"
+
+import pathlib
+import re
+import shutil
+import time
+import typing
+import numpy as np
+import cv2
+import open3d as o3d
+from rich.progress import Progress, track
+
+import znvis
+from znvis.rendering import Mitsuba
+
+
+class Headless_Renderer:
+    """
+    Main class to perform visualization.
+
+    Attributes
+    ----------
+    particles : list[znvis.Particle]
+            A list of particle objects to add to the visualizer.
+    counter : int
+            Internally stored counter to track which configuration is currently
+            being viewed.
+    """
+
+    def __init__(
+        self,
+        particles: typing.List[znvis.Particle],
+        vector_field: typing.List[znvis.VectorField] = None,
+        output_folder: typing.Union[str, pathlib.Path] = "./",
+        frame_rate: int = 24,
+        number_of_steps: int = None,
+        keep_frames: bool = True,
+        bounding_box: znvis.BoundingBox = None,
+        video_format: str = "mp4",
+        renderer_resolution: list = [4096, 2160],
+        renderer_spp: int = 64,
+        renderer: Mitsuba = Mitsuba(),
+        view_matrix: np.ndarray = np.array([[ 1,  0,  0, -100],
```

Comment:
Can we make this optional and add some classes to deal with different approaches?